### ISSUE TITLE
Where is the documentation for System.IdentityModel.Tokens.Jwt?

### ISSUE THREAD

**Issue by stamminator** [dotnet/docs#5321]
With Windows Identity Foundation (WIF) now integrated into .NET and the `System.IdentityModel.Tokens.Jwt` getting 8,000+ downloads every day, I don't understand why there isn't any official documentation. The [MSDN documentation is disgraceful](https://msdn.microsoft.com/en-us/library/system.identitymodel.tokens.jwtsecuritytoken(v=vs.114).aspx), offering nothing that IntelliSense doesn't already communicate, and [the official GitHub repo](https://github.com/AzureAD/azure-activedirectory-identitymodel-extensions-for-dotnet) for the NuGet package is a confusing mess that feels like trying to crack a walnut with a sledgehammer when trying to sift through the many projects just to find a simple, no-frills example.

For example, take a standard and common use case of needing to encrypt a JWT payload. The only usable solution I could find is [this Stack Overflow post(https://stackoverflow.com/questions/18223868/how-to-encrypt-jwt-security-token/44195678#44195678), and even that uses a hodgepodge of WIF 3.5 and .NET 4+ namespaces that feels more like a hack than a good solution.

JWTs are the standard now, and Microsoft Docs desperately needs dedicated documentation for developers. [There are others who are in need of this documentation as well](https://docs.microsoft.com/en-us/dotnet/framework/security/json-web-token-handler-api-reference). Please let us know what the plan is for this.

- Thanks

---
#### Document Details

⚠ *Do not edit this section. It is required for docs.microsoft.com ➟ GitHub issue linking.*

* ID: 06bdbd36-1298-419e-9ae8-e3582e804ced
* Version Independent ID: be11dcde-c28b-d40c-08c0-bc4d504c6df9
* Content: [System.IdentityModel Namespace](https://docs.microsoft.com/en-us/dotnet/api/system.identitymodel?view=netframework-4.7.2)
* Content Source: [xml/ns-System.IdentityModel.xml](https://github.com/dotnet/dotnet-api-docs/blob/master/xml/ns-System.IdentityModel.xml)
* Service: **.net**
* GitHub Login: @dotnet-bot
* Microsoft Alias: **v-jibran**

**Comment by mairaw** [comment]
Thanks for your feedback @stamminator. @brentschmaltz can you please comment on this issue? MSDN will no longer exist in a couple of months as well.

**Comment by mairaw** [comment]
This was moved to our backlog as a P3 work item @infoberg. We'll need @brentschmaltz to take a look at this request and take action.

**Comment by brentschmaltz** [comment]
@mairaw We are currently documenting here: https://github.com/AzureAD/azure-activedirectory-identitymodel-extensions-for-dotnet/wiki

**Comment by infoberg** [comment]
@brentschmaltz  please excuse me if I am missing something here, but do you call the release notes, links to the source code and 1 sample (which does not include JWE) a documentation?

**Comment by BoomerHudson** [comment]
Sent from Mail<https://go.microsoft.com/fwlink/?LinkId=550986> for Windows 10

**Comment by brentschmaltz** [comment]
@BoomerHudson @stamminator @infoberg noted, we haven't produced good docs.
https://github.com/AzureAD/azure-activedirectory-identitymodel-extensions-for-dotnet/wiki

This is a start.

Each time I am asked for sample code, I put it here: 
https://github.com/brentschmaltz/IdentityModel

**Comment by stamminator** [comment]
@brentschmaltz I appreciate the start, and the acknowledgement of the documentation issue. Is there a plan in place for getting proper documentation up to speed in general?

**Comment by brentschmaltz** [comment]
@stamminator We are at the planing stage for the doc work.

What would be most helpful for you?

**Comment by stamminator** [comment]
@brentschmaltz Well I only have my anecdotal example — I don't know what's being asked for in general — but I would say anything related to working with JWTs in a future-proof way. My use case was trying to encrypt a JWT using `System.IdentityModel.Tokens.Jwt`. The only [solution ](https://stackoverflow.com/a/44195678/4028303) I could find is not workable (as far as I know) for .NET Core, so when we move to that framework, we'll be forced to redo our entire JWT signing/handing implementation using something like `jose-jwt`.

**Comment by infoberg** [comment]
@brentschmaltz I would love to see samples and in-depth documentation on JWE

**Comment by brentschmaltz** [comment]
@stamminator @infoberg I am heading out for a week on vacation. I looked at the SO above, we can do better than that.
If you open an issue asking for a samples for different flavors of JWE (dir, keywrap, ect) we will get it done for 5.2.5 currently slated for Aug 10th.

**Comment by stamminator** [comment]
@brentschmaltz Thanks for your hard work. Enjoy vacation!

**Comment by mhamri** [comment]
@brentschmaltz I checked all the samples and in all of them they are working with azure! the sample for a simple web api with angular frontend that doesn't need azure backend is missing. special parts that i was looking for was token generation and user sign-in. in all the scenario, samples are delegating this tak to a third party issuer (azure/oAuth2) which both is not working for me.

**Comment by brentschmaltz** [comment]
@mhamri can you remind me of what you are looking for?

**Comment by laura-rodriguez** [comment]
Hi @brentsmaltz,
As per [OpenId Connect spec](https://openid.net/specs/openid-connect-core-1_0.html#IDTokenValidation):
_The `alg` value SHOULD be the default of RS256 or the algorithm sent by the Client in the id_token_signed_response_alg parameter during Registration._
Where exactly is this validated? In my case, I want to accept only tokens with `alg` equals to `RS256`.
Can you please point me where this validation is being made?

**Comment by Thiagopc** [comment]
Where have documentation System.IdentityModel.Tokens.Jwt?
A long time has passed!

**Comment by brentschmaltz** [comment]
@Thiagopc we started with this: https://github.com/AzureAD/azure-activedirectory-identitymodel-extensions-for-dotnet/wiki/Supported-Algorithms  

We are adding a section for Crypto extensibility.

**Comment by brentschmaltz** [comment]
@laura-rodriguez you are correct in the sense we don't explicitly check, we can add an option to check when calling OpenIdConnectProtocolValidator.

Although, the normal pattern of using discovery, makes me wonder why we need to do this?
